script.js

```diff
@@ -0,0 +1,245 @@
+        break;
+      case 'fun':
+        divButton.className += ' fun';
+        break;
+      case 'alt':
+        divButton.className += ' alt';
+        break;
+      case 'space':
+        divButton.className += ' space';
+        break;
+    }
+
+    // add on HTML document text on buttons
+    if (keyboard[j][i].length <= 2) {
+      divButton.innerHTML += `<span class="text" data-key="${keyboard[j][i][0]}">${keyboard[j][i][0]}</span><span class="textup" data-key="${keyboard[j][i][1]}">${keyboard[j][i][1]}</span>`;
+    } else {
+      divButton.innerHTML += `<span class="functionButton" data-key="${keyboard[j][i].key}">${keyboard[j][i].name}</span>`;
+    }
+    divRow.append(divButton);
+  }
+  keyBoard.append(divRow);
+}
+
+
+// hover buttons on shift
+textarea.addEventListener('keydown', function (event) {
+  if (event.key != 'Shift') return;
+  text.forEach((el) => el.style.display = 'none');
+  textUp.forEach((el) => el.style.display = 'inline');
+});
+
+textarea.addEventListener('keyup', function (event) {
+  if (event.key != 'Shift') return;
+  text.forEach((el) => el.style.display = 'inline');
+  textUp.forEach((el) => el.style.display = 'none');
+
+});
+
+// hover buttons on virtual keybord
+textarea.addEventListener('keydown', () => {
+  const element = document.querySelector('[data-key="' + event.key + '"]');
+  element.parentElement.classList.add('hover');
+
+ if (event.key === 'Tab') {
+   textarea.value += "\t";
+   event.preventDefault()
+ } else if (event.key === 'Enter') {
+   textarea.value += "\n";
+ }
+});
+
+textarea.addEventListener('keyup', () => {
+  const element = document.querySelector('[data-key="' + event.key + '"]');
+  element.parentElement.classList.remove('hover');
+});
+
+let text = document.querySelectorAll('.text');
+let textUp = document.querySelectorAll('.textup');
+text.forEach((el) => el.style.display = 'none');
+textUp.forEach((el) => el.style.display = 'inline');
+
+document.addEventListener('keydown', function (event) {
+  if (event.key != 'CapsLock') return;
+  let caps = event.getModifierState && event.getModifierState('CapsLock');
+
+  if (caps) {
+    text.forEach((el) => el.style.display = 'none');
+    textUp.forEach((el) => el.style.display = 'inline');
+  } else {
+    text.forEach((el) => el.style.display = 'inline');
+    textUp.forEach((el) => el.style.display = 'none');
+  }
+
+});
+
+const capslook = document.querySelector('.keybutton.caps');
+capslook.addEventListener('click', () =>{
+  if (text[0].style.display === 'inline') {
+    text.forEach((el) => el.style.display = 'none');
+    textUp.forEach((el) => el.style.display = 'inline');
+  } else {
+    text.forEach((el) => el.style.display = 'inline');
+    textUp.forEach((el) => el.style.display = 'none');
+  }
+
+});
+
+
+// adding to input by click translation
+
+
+keyBoard.addEventListener('mousedown', () => {
+  let element = '';
+  if (event.target.className === 'keyboard' || event.target.className === 'row') return;
```

Comment:
event.target.blablabla is too long
const target  = event.target;
or even better 
const { target }  = event;

and you this variable later